subsys/net/lib/http/http_server.c

```diff
@@ -0,0 +1,1187 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "headers/http_service.h"
+#include "headers/server_functions.h"
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+
+#include <zephyr/logging/log.h>
+#include <zephyr/net/http/service.h>
+#include <zephyr/net/tls_credentials.h>
+LOG_MODULE_REGISTER(net_http_server, CONFIG_NET_HTTP_SERVER_LOG_LEVEL);
+
+#define HTTP_SERVER_MAX_RESPONSE_SIZE CONFIG_NET_HTTP_SERVER_MAX_RESPONSE_SIZE
+#define HTTP_SERVER_MAX_URL_LENGTH    CONFIG_NET_HTTP_SERVER_MAX_URL_LENGTH
+#define HTTP_SERVER_MAX_FRAME_SIZE    CONFIG_NET_HTTP_SERVER_MAX_FRAME_SIZE
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/posix/sys/eventfd.h>
+
+static char url_buffer[CONFIG_NET_HTTP_SERVER_MAX_URL_LENGTH];
+static char http_response[CONFIG_NET_HTTP_SERVER_MAX_RESPONSE_SIZE];
+static const char *preface = "PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n";
+
+static struct http_parser_settings parserSettings;
+static struct http_parser parser;
+
+static struct arithmetic_result results[POST_REQUEST_STORAGE_LIMIT];
+static int results_count = 1;
+
+static unsigned char settings_frame[9] = {
+	0x00, 0x00, 0x00,        /* Length */
+	0x04,                    /* Type: 0x04 - setting frames for config or acknowledgment */
+	0x00,                    /* Flags: 0x00 - unused flags */
+	0x00, 0x00, 0x00, 0x00}; /* Reserved, Stream Identifier: 0x00 - overall connection */
+
+static unsigned char settings_ack[9] = {
+	0x00, 0x00, 0x00,        /* Length */
+	0x04,                    /* Type: 0x04 - setting frames for config or acknowledgment */
+	0x01,                    /* Flags: 0x01 - ACK */
+	0x00, 0x00, 0x00, 0x00}; /* Reserved, Stream Identifier */
+
+static const char content_404[] = {
+#ifdef INCLUDE_HTML_CONTENT
+#include "not_found_page.html.gz.inc"
+#endif
+};
+
+#define INVALID_SOCK -1
+
+bool has_upgrade_header = true;
+
+int http_server_init(struct http_server_ctx *ctx)
+{
+	int i = 1;
+	uint8_t af;
+	socklen_t len;
+	struct sockaddr_in6 addr_storage;
```

Comment:
There's a `struct sockaddr_storage` ;)

Reply:
Yes, very true. I just copied the sources without much changes so this need to be fixed.